### PR TITLE
[entropy_src/dv] Extend toggle coverage refines

### PR DESCRIPTION
**Pull request by andreaskurth**
- `entropy_src_xht_o`: This port was missing on the exclusions of XHT-related ports, although a exclusion for falling toggle coverage was previously in the UNR refine file.
- Some TL ports (see second commit for details) and `alert_rx_i[0]`: These ports are not exercised in block-level DV.  They were confirmed UNR in VCS, so this PR adds them to the UNR refine file as exclusions.

This PR contributes to resolving #16063.

With this, the toggle coverage of the `entropy_src` instance is 100%, but **dvsim still reports a  toggle coverage of only 80.3%** (both numbers with Xcelium). When checking the coverage report, I found that the toggle coverage of modules deeper in the hierarchy is evaluated as well (e.g., `entropy_src.u_reg.u_prim_reg_we_check.u_prim_onehot_check` -- although that is at 100%). I think we should only be taking the ports of the DUT into account for toggle coverage, right? Are we missing some configuration?

### CODE REVIEW

**Comment by andreaskurth** [comment]
@cindychip May I ask you to take a look at `entropy_src` toggle coverage concerning the questions in the PR?

File: hw/ip/entropy_src/dv/cov/entropy_src_UNR.vRefine [hunk 1]
```diff
@@ -24,10 +24,12 @@
...
+    <rule ccType="inst" domain="icc" entityName="entropy_src/&quot;tl_i.a_user.rsvd&quot;" entityType="toggle" excTime="1668165562" name="exclude" reviewer="unknown" user="1" vscope="default"></rule>
+    <rule ccType="inst" domain="icc" entityName="entropy_src/&quot;tl_o.d_param&quot;" entityType="toggle" excTime="1668165562" name="exclude" reviewer="unknown" user="1" vscope="default"></rule>
+    <rule ccType="inst" domain="icc" entityName="entropy_src/&quot;tl_o.d_sink&quot;" entityType="toggle" excTime="1668165562" name="exclude" reviewer="unknown" user="1" vscope="default"></rule>
+    <rule ccType="inst" domain="icc" entityName="entropy_src/&quot;alert_rx_i[0].ping_p&quot;" entityType="toggle" excTime="1668165562" name="exclude" reviewer="unknown" user="1" vscope="default"></rule>
+    <rule ccType="inst" domain="icc" entityName="entropy_src/&quot;alert_rx_i[0].ping_n&quot;" entityType="toggle" excTime="1668165562" name="exclude" reviewer="unknown" user="1" vscope="default"></rule>
```
Comment (sriyerg) [inline hw/ip/entropy_src/dv/cov/entropy_src_UNR.vRefine:32]: As mentioned in a comment in the other PR, these should have already been excluded by our common Xcelium coverage exclusion tcl script:
https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/xcelium/common_cov_excl.tcl#L6

It is worth investigating why it is not taking effect.

**Comment by sriyerg** [comment]
Is there an issue filed (tracking V3 for this block) for the removal of the `hw/ip/entropy_src/dv/cov/entropy_src_extht_exclusions.vRefine` file?

**Comment by sriyerg** [comment]
> Is there an issue filed (tracking V3 for this block) for the removal of the `hw/ip/entropy_src/dv/cov/entropy_src_extht_exclusions.vRefine` file?

@tjaychen 
Do you feel comfortable with this approach of excluding items that are going to be verified in V3 during V2, just so that we can artificially inflate the numbers for V2? We could file issues tracking the removal of those exclusions, but it feels risky to me.

**Comment by martin-lueker** [comment]
> > Is there an issue filed (tracking V3 for this block) for the removal of the `hw/ip/entropy_src/dv/cov/entropy_src_extht_exclusions.vRefine` file?
> 
> @tjaychen Do you feel comfortable with this approach of excluding items that are going to be verified in V3 during V2, just so that we can artificially inflate the numbers for V2? We could file issues tracking the removal of those exclusions, but it feels risky to me.

We can certainly do that.  

However I wouldn't call it artificial.  It is a feature that is most conclusively tied off at top level for this run, so there is some justification in not worrying about it for expediency.

**Comment by martin-lueker** [comment]
Issue #16276, created to track the removal of these exclusions.

**Comment by sriyerg** [comment]
> > > Is there an issue filed (tracking V3 for this block) for the removal of the `hw/ip/entropy_src/dv/cov/entropy_src_extht_exclusions.vRefine` file?
> > 
> > 
> > @tjaychen Do you feel comfortable with this approach of excluding items that are going to be verified in V3 during V2, just so that we can artificially inflate the numbers for V2? We could file issues tracking the removal of those exclusions, but it feels risky to me.
> 
> We can certainly do that.
> 
> However I wouldn't call it artificial. It is a feature that is most conclusively tied off at top level for this run, so there is some justification in not worrying about it for expediency.

I understand and agree with you on the de-prioritization of its verification. But, we _are_ going to verify it in the next stage. The concern here is lack of clear guidance on whether it is ok to exclude such features for the current milestone. If we collectively agree that it is ok to do so, then we also need to safeguard against leaving such temporarily added waivers in the codebase when we get to the next stage. The latter can be addressed by adding a new checklist item for V3.

**Comment by tjaychen** [comment]
i agree with @sriyerg. We do need to file an issue to track it (i think there might be one already, didn't you have one @martin-lueker ?).
Even if in this version of the top this feature is unused, we can't really declare a block level completion (D3/V3) unless we verified everything that is there.

**Comment by martin-lueker** [comment]
> i agree with @sriyerg. We do need to file an issue to track it (i think there might be one already, didn't you have one @martin-lueker ?). Even if in this version of the top this feature is unused, we can't really declare a block level completion (D3/V3) unless we verified everything that is there.

Wholly understood, and yes, the issue is #16276.

**Comment by cindychip** [comment]
> @cindychip May I ask you to take a look at `entropy_src` toggle coverage concerning the questions in the PR?

sorry totally missed the ping here. But we talked about it offline :)

**Comment by andreaskurth** [comment]
Closing as superseded by #16378

**Pull request closed without merging**